### DISCUSSION TITLE
Publish site as application on IIS

### DISCUSSION THREAD

**Discussion by stevewasielewski** [microsoft/fluentui-blazor#2675] [Q&A]
Here is the app.razor file:
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <base href="/">
    <link rel="stylesheet" href="app.css" />
    <link rel="stylesheet" href="LMCUGoals.styles.css" />

    <link href="_content/Microsoft.FluentUI.AspNetCore.Components/css/reboot.css" rel="stylesheet" />

    <HeadOutlet @rendermode="InteractiveServer" />
</head>

<body>
    <Routes @rendermode="InteractiveServer" />
    <script src="_framework/blazor.web.js"></script>
    <script src="_content/Microsoft.FluentUI.AspNetCore.Components/Microsoft.FluentUI.AspNetCore.Components.lib.module.js" type="module" async></script>

</body>

</html>
and after I publish it - I'm getting 404 errors on finding the files.  Any ideas?

I've used Mudblazor in the past with no issues. I've trying changing the base href too.

Thanks

**Comment by vnbaaij** [comment]
Which version of the library are you using?

If you use the application generated scoped css file, you don't need to include reboot.

Also, you do not have to include the library's js file. This is loaded automatically.

Maybe try with a template generated site first?

**Comment by stevewasielewski** [comment]
I just took the template project and try to install on IIS as application and still getting the errors.

Version: 4.10.0

**Comment by vnbaaij** [comment]
Please post a zip of the project you have created. Otherwise this is going to stay a guessing game for us.

**Comment by stevewasielewski** [comment]
Will do

**Comment by MarvinKlein1508** [comment]
We don't have any of this issues on our production app which is also being hosted in IIS. Pretty sure you did something wrong in your project.

**Comment by vnbaaij** [comment]
I'm moving this to the discussion, as I don't think this is an issue in the library but more in configuration,

**Comment by stevewasielewski** [comment]
Has anyone had any issue with a Blazor Server 8 app running on IIS as an application under a default site?

**Comment by NabaviFard-Mahdi** [comment]
Have you moved pages in the class library?
Or you use the structure of components/pages in the main project

**Comment by samuelkim-scz** [comment]
I'm having similar issues.  When hosting from its own Web Site, then all works well.  When hosting as an Application (and Virtual Directory) under Default Web Site then the error occurs.  

When under its own Web Site and using "~/" as base href, then it also does not work.

**Comment by marrale** [comment]
I'm having same issue, loading the "reboot.css" file in a site hosted as application in IIS. I have a .net 9 WebApp with global render mode "InteractiveServer". And I'm using the "Microsoft.FluentUI.AspNetCore.Components" v4.11.9.
This is my "App.razor" where I set the "base href" to my sub path "mySite":
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <base href="/mySite/" />
    <link rel="stylesheet" href="@Assets["app.css"]" />
    <link rel="stylesheet" href="@Assets["MyFluentUiSite.styles.css"]" />
    <ImportMap />
    <link rel="icon" type="image/x-icon" href="favicon.png" />
    <HeadOutlet @rendermode="InteractiveServer" />
</head>
<body>
    <Routes @rendermode="InteractiveServer" />
    <script src="_framework/blazor.web.js"></script>
</body>
</html>
```

All other pages load correctly with the url "https://myhost/mySite/..."
Only the "reboot.css" loads wrongly with "https://myhost/..." skipping the "mySite" sub path.

![image](https://github.com/user-attachments/assets/0b415c6a-5875-4c07-b684-aeb452d3b465)

Any idea what I'm doing wrong?

**Comment by MarvinKlein1508** [comment]
@marrale I think this might be related to the import statement within the `app.css`

Could you try removing the line 
```css
@import '_content/Microsoft.FluentUI.AspNetCore.Components/css/reboot.css';
```

From your `app.css` and add 
```razor
<link href="@Assets["/_content/Microsoft.FluentUI.AspNetCore.Components/css/reboot.css"]" rel="stylesheet" />
```

before:
```razor
<link rel="stylesheet" href="@Assets["app.css"]" />
```

in your `App.razor` 

I may not encounter this issue because I never changed the `base` tag in my projects from `<base href="/" />`

**Comment by marrale** [comment]
@MarvinKlein1508 thanks for your hint, now it works as expected, I needed a small change from your suggestion, I removed the "/" before "_content":
```html
<link href="@Assets["_content/Microsoft.FluentUI.AspNetCore.Components/css/reboot.css"]" rel="stylesheet" />
```

This is now my "App.razor" with a dynamic base path, this should work with any installation:
```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <base />
    <script>
        var path = window.location.pathname.split('/');
        var base = document.getElementsByTagName('base')[0];
        if (window.location.host.includes('localhost')) {
            base.setAttribute('href', '/');
        } else if (path.length > 2) {
            base.setAttribute('href', '/' + path[1] + '/');
        } else if (path[path.length - 1].length != 0) {
            window.location.replace(window.location.origin + window.location.pathname + '/' + window.location.search);
        }
    </script>
    <link rel="stylesheet" href="@Assets["app.css"]" />
    <link href="@Assets["_content/Microsoft.FluentUI.AspNetCore.Components/css/reboot.css"]" rel="stylesheet" />
    <link rel="stylesheet" href="@Assets["MyFluentUiSite.styles.css"]" />
    <ImportMap />
    <link rel="icon" type="image/x-icon" href="favicon.png" />
    <HeadOutlet @rendermode="InteractiveServer" />
</head>

<body>
    <Routes @rendermode="InteractiveServer" />
    <script src="_framework/blazor.web.js"></script>
</body>

</html>
```

And this is the commented out first line in "app.css":
`
/* @import '/_content/Microsoft.FluentUI.AspNetCore.Components/css/reboot.css'; */
`

**Comment by MarvinKlein1508** [comment]
thanks for verifying @marrale 

@vnbaaij is there any reason the import is being done in `app.css`? If not I would suggest updating the templates to move reboot into `App.razor`. I would do it myself but I have never worked with creating templates before.

**Reply by vnbaaij** [reply]
It is done in `app.css` so in most cases a developer does not have to think about it an 'it just works'.

It should be possible to have an `@import` in a CSS file when using a base tag, I'd think.

But instead of changing the template, it might be more convenient (as it works normally in almost all cases) to document this setup in the https://www.fluentui-blazor.net/CodeSetup#reboot section. @MarvinKlein1508 may I ask you to contribute to that?

**Reply by MarvinKlein1508** [reply]
Will create a PR later :)
